src/Core/Utility.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * @constructor
+ */
+function List () {
+	this.next = this
+	this.prev = this
+	this.length = 0
+}
+List.prototype = {
+	/**
+	 * @param {Element} element
+	 * @param {Element} sibling
+	 * @return {Element}
+	 */
+	insert: function insert (element, sibling) {
+		element.next = sibling
+		element.prev = sibling.prev
+		sibling.prev.next = element
```

Comment:
should be `element.prev.next` (equivalent, but less confusing)

Reply:
Why is it confusing?

Reply:
Because it looks like you're trying to set `prev.next` on `sibling` but it actually only sets it on `element`

```
element.prev = sibling.prev
sibling.prev.next = element
sibling.prev = element // prev.next above is 'ignored' because it mutates the binding on element
```

Reply:
I think its less error prone as it is since element is a new item while sibling is an item that already exist in the List, if we switched the position of these two statement with each other the current implementation would still work while `element.prev.next` would fail.

Reply:
you could argue that the proposed version would allow the switching of the last two assignments, but I digress. Writing to and then de-referencing an object immediately is non-intuitive.